### ISSUE TITLE
HB fit ignores seed= and seeds NUTS from wall-clock time

### ISSUE THREAD

**Issue by hanhou** [AllenNeuralDynamics/aind-dynamic-foraging-bfm-dispatcher#108]
## Context

Found while running the first HB job on Beaker (#107). The launch passed `seed=0`, but the
HB fit seeded itself from wall-clock time instead.

From the smoke job's log (Beaker experiment `01M1JWY7VNB9F73BS14679K78Q`):

```
[root][INFO] - No seed set for ModelTrainer, using 1788414615
```

## Root cause

`code/config/model/hb_hattori.yaml` has no `seed:` key. Every other model config in the
group does:

| config | has `seed: ${seed}` |
|---|---|
| `baseline_grn.yaml` | yes (l.7) |
| `baseline_rl.yaml` | yes (l.40) |
| `baseline_rl_bari.yaml` | yes (l.29) |
| `baseline_rl_ctt.yaml` | yes (l.26) |
| `baseline_rl_hattori.yaml` | yes (l.26) |
| `baseline_rl_losscounting.yaml` | yes (l.27) |
| `baseline_rl_rw.yaml` | yes (l.34) |
| `disrnn.yaml` | yes (l.114) |
| `gru.yaml` | yes (l.62) |
| `gru_scaling.yaml` | yes (l.113) |
| **`hb_hattori.yaml`** | **no** |

Hydra `instantiate` therefore calls `HBTrainer(seed=None)`, and
`ModelTrainer.__init__` (wrapper `code/base/interfaces.py:30-33`) falls back to
`int(time.time())`.

`HBTrainer` itself is wired correctly and needs no change — it accepts `seed` and derives
both PRNG keys from it (`code/model_trainers/hb_trainer.py:447` and `:590`,
`jax.random.PRNGKey(int(self.seed or 0))`). The config is the only gap.

## Why it matters

Both NUTS keys — the population fit and the held-out adaptation/scoring draws — come from
that seed, so **an HB rung cannot be reproduced from its recorded config.** Study 08 exists
to compare HB against study 01's GRU at matched D, where the GRU's seed spread of ~0.0004
is the stated yardstick for whether a difference means anything. An HB number with an
uncontrolled seed cannot be placed on that scale, and a re-run to check it will not return
the same value.

Mitigating detail: the resolved seed *is* persisted into the fit artifact
(`hb_trainer.py:405` writes `"seed": self.seed`), so already-produced runs are
attributable after the fact — recoverable, not controllable.

## Done when

- [ ] `hb_hattori.yaml` carries `seed: ${seed}`
- [ ] A launch with `seed=0` logs `Using seed 0 for ModelTrainer`, not `No seed set`
- [ ] The D≈30 rung for #107 runs with the fix in place

## Notes

Verified from the config files and the wrapper source at
dispatcher `d49b4a8` / wrapper `d7309dd`. Not yet verified: whether two runs at the same
seed reproduce bit-for-bit on the same GPU — the fix makes the seed *controlled*, and
NumPyro/XLA determinism across identical devices is a separate question (see #101 for the
cross-stack version of it).

**Comment by hanhou** [comment]
Sharper root cause than filed, plus confirmation the fix works on a node.

**This is a regression, not an original gap.** `hb_hattori.yaml` never had a `seed:` key —
not even at `8defd18`, the dispatcher SHA the 2026-08-29 HPC ladder ran from. Yet those runs
logged `Using seed 0 for ModelTrainer` and were named `hb-one_stage-D10-s0`. The seed reached
the trainer through a different route: HB had its own entrypoint, `run_hb.py`, with real
seed plumbing — `parser.add_argument("--seed", type=int, default=0)`,
`jax.random.PRNGKey(args.seed)`, `seed=args.seed` into the trainer, and the value stamped
into both the W&B config and the run name.

`906bc32` ("refactor(study-08): run HB through run_hpc instead of a parallel entrypoint",
2026-08-30) deleted that entrypoint. The trainer is now built purely by
`instantiate(hydra_config.model)` — `training_runner.py:184`/`:326` — and nothing supplies a
compensating `seed: ${seed}` in the model config, so the seed silently stopped arriving.
(`training_runner.py:134` does read the global seed, but that is inside the auto held-out
fine-tune config for the GRU/disRNN path, unrelated to trainer construction.)

The regression is visible in W&B either side of that commit:

| run | dispatcher | log line |
|---|---|---|
| `hb-one_stage-D10-s0` (2026-08-29, HPC) | `8defd18`, pre-refactor | `Using seed 0 for ModelTrainer` |
| `one-stage-ladder-20260902-224741-…` (2026-09-03, Beaker) | `d49b4a8`, post-refactor | `No seed set for ModelTrainer, using 1788414615` |

**Fix verified end to end.** Three replicate launches with `DISPATCHER_REF` pinned to the
fix commit `26b579d` all log:

```
[root][INFO] - Using seed 0 for ModelTrainer
```

That closes the second box in *Done when*. Note what the fix does **not** cover: the three
2026-08-29 HPC D10 runs were all at seed 0 on the same cohort and returned 0.7087584 /
0.7092775 / 0.7095207 — a spread of 7.6e-4, roughly twice the GRU's ~4e-4 seed spread that
study 08 uses as its yardstick. Their sampler settings are not recorded in W&B, so that
spread is not yet attributable to nondeterminism rather than to differing settings. The three
replicates above are the controlled version of that experiment and will settle it.

Also worth noting for whoever picks this up: the refactor traded an informative run name
(`hb-one_stage-D10-s0` — estimator, D, seed) for the generic Hydra template, which is why
post-refactor runs read `mice_snapshot_multisubject_hb_one_stage_multisubject_n10subj`. That
is cosmetic and deliberately not being fixed here.
